### ISSUE TITLE
how to avoided overfit?

### ISSUE THREAD

**Issue by jinfagang** [zylo117/Yet-Another-EfficientDet-Pytorch#303]
Training on a single class dataset, set the category_id to 0 for all. Also edit obj_list in yml.

the model totally overfit in such situation, how to avoid it?

![image](https://user-images.githubusercontent.com/21303438/82036696-caa07280-96d3-11ea-902b-a73a49b41bb6.png)

Don't even know why it so hard to train when only changed classes.

**Comment by zylo117** [comment]
Maybe just try one-based indexing.

**Comment by zylo117** [comment]
it's hard to debug when you modify the code. can you open a branch?

**Comment by jinfagang** [comment]
doesn't change anything, simple removed `category_id - 1` due to my annotations index from 0.

**Comment by jinfagang** [comment]
I forced set category_id to all 0 for single class, it also overfited. (I have changed effcientdet num_classes as well)

**Comment by Cli98** [comment]
> doesn't change anything, simple removed `category_id - 1` due to my annotations index from 0.

@jinfagang Don't do this. If you set with 0-index, then when you compute result, zylo117's code will automatically -1 for you. So your label completely messed up. If you keep with 0-index, then modify inference code instead.

**Comment by jinfagang** [comment]
@Cli98  What if I make coco_annotate.json index from 1, then assign all label id to 1, is that doable? (I just need to try single class this time, all target in my data can be treated as single class)

**Comment by Cli98** [comment]
> @Cli98 What if I make coco_annotate.json index from 1, then assign all label id to 1, is that doable? (I just need to try single class this time, all target in my data can be treated as single class)

@jinfagang Just index your json file from 1, Do not worry about anything else. I have exactly same worry as you at beginning. After I check the code I just figure out that's a waste of time lol.
